indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java

```diff
@@ -0,0 +1,1021 @@
+        {
+          // NOOP
+        }
+
+        @Override
+        public void close(TaskAttemptContext context) throws IOException, InterruptedException
+        {
+          // NOOP
+        }
+      };
+    }
+
+    @Override
+    public void checkOutputSpecs(JobContext context) throws IOException, InterruptedException
+    {
+      // NOOP
+    }
+
+    @Override
+    public OutputCommitter getOutputCommitter(final TaskAttemptContext context)
+        throws IOException, InterruptedException
+    {
+      return new OutputCommitter()
+      {
+        @Override
+        public void setupJob(JobContext jobContext) throws IOException
+        {
+
+        }
+
+        @Override
+        public void setupTask(TaskAttemptContext taskContext) throws IOException
+        {
+
+        }
+
+        @Override
+        public boolean needsTaskCommit(TaskAttemptContext taskContext) throws IOException
+        {
+          return taskContext.getConfiguration().get(PUBLISHED_SEGMENT_KEY) != null;
+        }
+
+        @Override
+        public void commitTask(final TaskAttemptContext taskContext) throws IOException
+        {
+          final Progressable commitProgressable = new Progressable()
+          {
+            @Override
+            public void progress()
+            {
+              taskContext.progress();
+            }
+          };
+          final HadoopDruidConverterConfig config = converterConfigFromConfiguration(taskContext.getConfiguration());
+          final String finalSegmentString = taskContext.getConfiguration().get(PUBLISHED_SEGMENT_KEY);
+          if (finalSegmentString == null) {
+            throw new IOException("Could not read final segment");
+          }
+          final DataSegment newSegment = HadoopDruidConverterConfig.jsonMapper.readValue(
+              finalSegmentString,
+              DataSegment.class
+          );
+          log.info("Committing new segment");
+          taskContext.progress();
+
+          final FileSystem fs = taskContext.getWorkingDirectory().getFileSystem(taskContext.getConfiguration());
+          final Path taskAttemptDir = getTaskPath(
+              context.getJobID(),
+              context.getTaskAttemptID(),
+              taskContext.getWorkingDirectory()
+          );
+          final Path taskAttemptFile = new Path(taskAttemptDir, DATA_FILE_KEY);
+          final Path taskAttemptSuccess = new Path(taskAttemptDir, DATA_SUCCESS_KEY);
+          try (final OutputStream outputStream = fs.create(taskAttemptFile, false, 1 << 10, commitProgressable)) {
+            outputStream.write(HadoopDruidConverterConfig.jsonMapper.writeValueAsBytes(newSegment));
+          }
+
+          fs.create(taskAttemptSuccess, false).close();
+
+          taskContext.progress();
+          taskContext.setStatus("Committed");
+        }
+
+        @Override
+        public void abortTask(TaskAttemptContext taskContext) throws IOException
+        {
+          log.warn("Aborting task");
```

Comment:
This is only logs for the OutputCommitter. Ideally you would cleanup any intermediate stuff here, but since our intermediate files are handled in the Mapper, there is nothing here to cleanup. This *should* be surrounded by other log messages indicating the reason for the abort and other informations outside the OutputCommitter